src/ui.rs

```diff
@@ -0,0 +1,335 @@
+        &text,
+        screen_width * GENERATIONS_POS_X,
+        screen_height * GENERATIONS_POS_Y,
+        TEXT_SIZE,
+        BLACK,
+    );
+
+    (offset_x, offset_y, sq_size)
+}
+
+/// Handles the edit mode where users can add or remove cells.
+fn run_edit_mode(game: &mut Game, offset_x: &f32, offset_y: &f32, sq_size: &f32) {
+    show_edit_hud();
+
+    if is_mouse_button_pressed(MouseButton::Left) {
+        let (mouse_x, mouse_y) = mouse_position();
+
+        let grid_x = ((mouse_x - offset_x) / sq_size).floor() as i32;
+        let grid_y = ((mouse_y - offset_y) / sq_size).floor() as i32;
+
+        if grid_x >= 0 && grid_x < SQUARES as i32 && grid_y >= 0 && grid_y < SQUARES as i32 {
+            if game.is_cell_alive(grid_x, grid_y) {
+                game.remove_cell(grid_x, grid_y);
+            } else {
+                game.add_cell(grid_x, grid_y);
+            }
+        }
+    }
+}
+
+/// Displays the HUD for edit mode, showing instructions to the user.
+fn show_edit_hud() {
+    let screen_width = screen_width();
+    let screen_height = screen_height();
+
+    draw_text(
+        "Edit Mode",
+        screen_width * TITLE_POS_X,
+        screen_height * TITLE_POS_Y,
+        TITLE_SIZE,
+        BLACK,
+    );
+    draw_text(
+        "Click on a live cell to kill it",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y - 15.,
+        TEXT_SIZE,
+        BLACK,
+    );
+    draw_text(
+        "Click on a dead cell to make it alive",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y,
+        TEXT_SIZE,
+        BLACK,
+    );
+    draw_text(
+        "Press E to exit Edit Mode",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y + 15.,
+        TEXT_SIZE,
+        BLACK,
+    );
+}
+
+/// Runs the game in paused mode, displaying the pause HUD.
+fn run_paused_mode() {
+    show_pause_hud();
+}
+
+/// Displays the HUD for paused mode, showing instructions for the user.    
+fn show_pause_hud() {
+    let screen_width = screen_width();
+    let screen_height = screen_height();
+
+    draw_text(
+        "Game Paused",
+        screen_width * TITLE_POS_X,
+        screen_height * TITLE_POS_Y,
+        TITLE_SIZE,
+        BLACK,
+    );
+    draw_text(
+        "Press E to enter Edit Mode",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y,
+        TEXT_SIZE,
+        BLACK,
+    );
+    draw_text(
+        "Press X to exit the game",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y + 15.,
+        TEXT_SIZE,
+        BLACK,
+    );
+    draw_text(
+        "Press R to restart the game with a random seed",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y + 30.,
+        TEXT_SIZE,
+        BLACK,
+    );
+}
+
+/// Runs the game in playing mode, where time passes and the game is updated.
+fn run_playing_mode(game: &mut Game, generations_passed: &mut i32) {
+    sleep(Duration::from_secs(1));
```

Comment:
There is no problem adding an sleep, but we could move it with a `FPS` variable, also the handling of user input shouldn't be stopped as it makes the game difficult to use, it should only affect the drawing

Reply:
Done!